packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -614,8 +614,11 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
       } else {
         _shouldShowBackArrow ??= Navigator.canPop(context);
         if (_shouldShowBackArrow) {
+          IconData backIcon = Theme.of(context).platform == TargetPlatform.iOS
+              ? Icons.arrow_back_ios
+              : Icons.arrow_back;
```

Comment:
Please use a `switch` statement.  That will make the analyze complain when we add a new target platform and someone will need to think about which icon it should have.